slippymap/slippymap.go

```diff
@@ -74,29 +73,23 @@ func (sm *SlippyMap) Draw(screen *ebiten.Image) {
 	screen.DrawImage(sm.zoomPrevLevelImg, nil)
 
 	// draws all tiles onto screen
-	wg := sync.WaitGroup{}
 
-	for _, tile := range sm.tiles {
-		wg.Add(1)
-		go func(t *MapTile) {
-			dio := &ebiten.DrawImageOptions{}
+	for _, t := range sm.tiles {
+		dio := &ebiten.DrawImageOptions{}
 
-			// move the image where it needs to be in the window
-			dio.GeoM.Translate(float64((*t).offsetX), float64((*t).offsetY))
+		// move the image where it needs to be in the window
+		dio.GeoM.Translate(float64((*t).offsetX), float64((*t).offsetY))
 
-			// adjust transparency (for fade-in of tiles)
-			dio.ColorM.Scale(1, 1, 1, (*t).alpha)
+		// adjust transparency (for fade-in of tiles)
+		dio.ColorM.Scale(1, 1, 1, (*t).alpha)
 
-			// draw the tile
-			screen.DrawImage(t.img, dio)
+		// draw the tile
+		screen.DrawImage(t.img, dio)
 
-			// debugging: print the OSM tile X/Y/Z
-			dbgText := fmt.Sprintf("%d/%d/%d", (*t).osmX, (*t).osmY, (*t).zoomLevel)
-			ebitenutil.DebugPrintAt(screen, dbgText, (*t).offsetX, (*t).offsetY)
-			wg.Done()
-		}(tile)
+		// debugging: print the OSM tile X/Y/Z
+		dbgText := fmt.Sprintf("%d/%d/%d", (*t).osmX, (*t).osmY, (*t).zoomLevel)
+		ebitenutil.DebugPrintAt(screen, dbgText, (*t).offsetX, (*t).offsetY)
```

Comment:
`DebugPrintAt` can't be called concurrently, it has map reads and modifications that need to be protected by a mutex. I got regular panics on application load. 

I've removed the goroutine from here because I couldn't spot any tangible performance change on my laptop. I am unsure if there's a reason goroutines were used here or not. Alternatives include removing the call to `DebugPrintAt` entirely, or moving the call to be outside of the goroutine.

Reply:
It's mostly there for debugging and development -
I didn't plan for it to make it through to production. I'm OK with it being moved outside the goroutine.

Reply:
Also @bluntelk did the goroutine stuff so tagging him in.

Reply:
if it is of little benefit, by all means, nuke it.

I think this was the first go routine I put in trying to speed things up.